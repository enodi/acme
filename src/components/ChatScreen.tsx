import React from "react";

import InputField from "./InputField";
import ChatConversation from "./ChatConversation";
import { userDataFormat } from "../models";

interface ChatScreenProps {
  selectedItem: userDataFormat;
  conversation: userDataFormat;
  handleKeyDown: (event: any) => void;
  onTextChange: (event: React.ChangeEvent<HTMLInputElement>) => void;
  messageText: string;
}

const ChatScreen: React.FC<ChatScreenProps> = ({
  selectedItem,
  conversation,
  handleKeyDown,
  onTextChange,
  messageText,
}) => (
  <section className="chat-screen">
    {conversation.messages && conversation.messages.length > 0 ? (
      <ChatConversation conversation={conversation} />
    ) : (
      <div className="chat-screen__info">
        <h1>Select a conversation</h1>
        <p>
          Start by selecting a conversation or searching for someone specific
        </p>
      </div>
    )}
    {selectedItem?.isSelected && (
      <InputField
        handleKeyDown={handleKeyDown}
        onTextChange={onTextChange}
        messageText={messageText}
      />
    )}
  </section>
);

export default ChatScreen;
