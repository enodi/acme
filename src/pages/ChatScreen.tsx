import React from "react";

const ChatScreen: React.FC = () => <h1>hello world</h1>;

export default ChatScreen;
