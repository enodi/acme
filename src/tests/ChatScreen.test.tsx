import React from "react";
import { shallow } from "enzyme";

import ChatScreen from "../components/ChatScreen";
import { selectedItem } from "./fixtures";

let handleKeyDown, onTextChange, wrapper: any;
beforeEach(() => {
  handleKeyDown = jest.fn();
  onTextChange = jest.fn();
  wrapper = shallow(
    <ChatScreen
      handleKeyDown={handleKeyDown}
      selectedItem={selectedItem}
      conversation={selectedItem}
      onTextChange={onTextChange}
      messageText={selectedItem.messages[0]}
    />
  );
});

test("should render ChatScreen component correctly", () => {
  expect(wrapper).toMatchSnapshot();
});
