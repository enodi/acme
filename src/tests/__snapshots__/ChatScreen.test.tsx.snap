// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ChatScreen component correctly 1`] = `
<section
  className="chat-screen"
>
  <ChatConversation
    conversation={
      Object {
        "email": "krotte0@webmd.com",
        "firstName": "Kirby",
        "id": 1,
        "isFavorite": true,
        "isSelected": true,
        "lastName": "Rotte",
        "messages": Array [
          "Lorem ipsum dolor sit amet, consectetur adipiscing elit",
        ],
      }
    }
  />
  <InputField
    handleKeyDown={[MockFunction]}
    messageText="Lorem ipsum dolor sit amet, consectetur adipiscing elit"
    onTextChange={[MockFunction]}
  />
</section>
`;
